lib/acts_as_taggable_on/taggable/core.rb

```diff
@@ -28,12 +29,16 @@ def initialize_acts_as_taggable_on_core
             has_many context_taggings, -> { includes(:tag).order(taggings_order).where(context: tags_type) },
                      as: :taggable,
                      class_name: 'ActsAsTaggableOn::Tagging',
-                     dependent: :destroy
+                     dependent: :destroy,
+                     after_add: :dirtify_tag_list,
+                     after_remove: :dirtify_tag_list
 
             has_many context_tags, -> { order(taggings_order) },
                      class_name: 'ActsAsTaggableOn::Tag',
                      through: context_taggings,
                      source: :tag
+
+            attribute "#{tags_type.singularize}_list".to_sym, ActiveModel::Type::Value.new
```

Comment:
This line makes `taggable_obj.attribute_names` changed. Specs failed in the following case, using `json.extract! @item, *@item.attribute_names` as response. 

I saw you comment this PR will be released as the version 6.0.0 and allow breaking changes, but the impact of this change, if you don't expect, is very big.

What do you think about this case?

Reply:
@Fodoj

Reply:
Not sure I understand the issue. Could you provide a failing spec?

Though I think if it's 6.0.0 release, then breaking changes are totally ok, just repo owner has to update readme before reasling this new version.

Reply:
Failing is request spec with the [rspec-json_matcher gem](https://github.com/r7kamura/rspec-json_matcher). I copy and modify some from the real codes and output of running rspec:

```ruby
# request spec
it { is_expected.to be_json_as [product: expected_product_response] }
```

```
# output
expected to match:

  {
    :product => {
      :id          => Integer < Numeric,
      :user_id     => Integer < Numeric,
      :product_id  => Integer < Numeric,
      :created_at  => String < Object,
      :updated_at  => String < Object,
    }
  }

actual:
  {
    :product => {
      "id"          => 2,
      "user_id"     => 11,
      "product_id"  => 216,
      "tag_list"    => []
      "created_at"  => "2018-06-05T17:36:33+09:00",
      "updated_at"  => "2018-06-05T17:36:33+09:00"
    }
  }


reason: product "tag_list"
```

Reply:
The case mentioned above is with the rspec-json_matcher gem, but if not using the rspec-json_matcher gem, the test suites testing api response would all fail because of the change.

Reply:
In my opinion this change is harmful #1064